x-pack/plugins/fleet/server/services/agents/versions.ts

```diff
@@ -51,7 +55,7 @@ export const getAvailableVersions = async (cached = true): Promise<string[]> =>
       .sort((a: any, b: any) => (semverGt(a, b) ? -1 : 1));
     versionsToDisplay = uniq(versions) as string[];
 
-    if (!config?.internal?.onlyAllowAgentUpgradeToKnownVersions) {
+    if (!config?.internal?.onlyAllowAgentUpgradeToKnownVersions && includeCurrentVersion) {
```

Comment:
Should we change this so the default for `includeCurrentVersion is `!config?.internal?.onlyAllowAgentUpgradeToKnownVersions`?

Reply:
Addressed in [bd768ab](https://github.com/elastic/kibana/pull/166811/commits/bd768abb1bb54c9c227b0c3078da562117d1d8fd).